// app/api/aa-lib.js
const axios = require("axios")
const errCodeMap = require("./aa-error-code")

const axClient = axios.create({
  baseURL: "http://localhost:8081",
  timeout: 1000,
  headers: {
    // "Content-Encoding": "gzip",
  },
  transformResponse: [
    (resp) => {
      try {
        return JSON.parse(resp)
      } catch (err) {
        throw {
          code: 500,
          msg: "服务器错误",
          data: null,
        }
      }
    },
    (resp) => {
      let code = resp.code
      if (code !== 0) {
        throw {
          code: code,
          msg: errCodeMap.get(code),
          data: resp.data,
        }
      }
      return resp.data
    },
  ],
})

/*
 *
 * api function usage:
 *   api.GetUser()
 *     .then((data) => {
 *       console.log(data)
 *     })
 *     .catch((err) => {
 *       console.log(err)
 *     })
 *
 */

module.exports = {
  ax(path, postData) {
    return axClient.post(path, postData).then((resp) => {
      return resp.data
    })
  },

  /*
   * usage:
   *   function GetUser() {
   *     return mock(0, {
   *       name: "Alice",
   *     })
   *   }
   */
  mock(code, data) {
    return new Promise((resolve, reject) => {
      return resolve({
        code,
        data,
      })
    })
  },
}
